include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -1028,6 +1056,106 @@ struct __remove_fn
+             std::indirect_equivalence_relation<std::projected<std::ranges::iterator_t<_InRange>, _Proj>>
+                _Comp = std::ranges::equal_to>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>>
+        && std::ranges::sized_range<_InRange> && std::ranges::sized_range<_OutRange>
+        && std::indirectly_copyable<std::ranges::iterator_t<_InRange>, std::ranges::iterator_t<_OutRange>>
+
+    std::ranges::unique_copy_result<std::ranges::borrowed_iterator_t<_InRange>, std::ranges::borrowed_iterator_t<_OutRange>>
+    operator()(_ExecutionPolicy&& __exec, _InRange&& __in_r, _OutRange&& __out_r, _Comp __comp = {}, _Proj __proj = {}) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_unique_copy(__dispatch_tag,
+            std::forward<_ExecutionPolicy>(__exec), std::forward<_InRange>(__in_r), std::forward<_OutRange>(__out_r),
+            __comp, __proj);
+    }
+};//__unique_fn
+} //__internal
+
+inline constexpr __internal::__unique_copy_fn unique_copy;
+
+// [alg.reverse]
+
+namespace __internal
+{
+struct __reverse_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>>
+        && std::permutable<std::ranges::iterator_t<_R>> && std::ranges::sized_range<_R>
+
+    std::ranges::borrowed_iterator_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r) const
+    {
+        const auto __n = std::ranges::size(__r);
+        const auto __n_2 = __n / 2;
+        auto __r1 = std::ranges::take_view(__r, __n_2);
+        auto __r2 = std::ranges::take_view(std::ranges::reverse_view(__r), __n_2);
+
+        oneapi::dpl::ranges::swap_ranges(std::forward<_ExecutionPolicy>(__exec), std::move(__r1),std::move(__r2));
+        return {std::ranges::begin(__r) + __n};
+    }
+
+}; //__reverse_fn
+} //__internal
+
+inline constexpr __internal::__reverse_fn reverse;
+
+namespace __internal
+{
+struct __reverse_copy_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _InRange,
+             std::ranges::random_access_range _OutRange>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>>
+        && std::ranges::sized_range<_InRange> && std::ranges::sized_range<_OutRange>
+        && std::indirectly_copyable<std::ranges::iterator_t<_InRange>, std::ranges::iterator_t<_OutRange>>
+
+    std::ranges::reverse_copy_result<std::ranges::borrowed_subrange_t<_InRange>,
+                                     std::ranges::borrowed_subrange_t<_OutRange>>
+    operator()(_ExecutionPolicy&& __exec, _InRange&& __in_r, _OutRange&& __out_r) const
+    {
+        return oneapi::dpl::ranges::copy(std::forward<_ExecutionPolicy>(__exec), std::ranges::reverse_view(__in_r),
+            std::forward<_OutRange>(__out_r));
```

Comment:
`__pattern_reverse_copy` from the device backend also uses an optimized `__reverse_copy` routine. Let's reuse it.

Reply:
Lets wait an update from C++ committee. There is a chance that `ranges::reverse_copy` with policy will be removed from the parallel algo set.

Reply:
Update: The question was proceed with C++ committee and accepted into the C++ standard.
So,  `ranges::reverse_copy` should return 3 iterators: `the last` in the input ranges, `a last position` in the input ranges, `the last` in the output range.

Reply:
Closing as reverse and reverse copy are moved into https://github.com/uxlfoundation/oneDPL/pull/2363.